packages/stream/StreamClient.ts

```diff
@@ -708,3 +747,59 @@ async function ata(mint: PublicKey, account: PublicKey) {
     account
   );
 }
+
+function isAnchorWallet(wallet: Keypair | Wallet): wallet is Wallet {
+  //magic happens here
+  return (<Wallet>wallet).signTransaction !== undefined;
+}
+
+interface BatchItem {
+  recipient: string;
+  tx: web3.Transaction;
+}
+
+interface BatchItemSuccess extends BatchItem {
+  signature: string;
+}
+
+interface BatchItemError extends BatchItem {
+  error: string;
+}
+
+async function signAllTransactionWithRecipients(
+  sender: Keypair | Wallet,
+  batch: BatchItem[]
+) {
+  let signed_batch: BatchItem[] = [];
+  const isKeypair = sender instanceof Keypair;
+  const isWallet = isAnchorWallet(sender);
+
+  if (!isKeypair && !isWallet) return signed_batch;
+
+  if (isKeypair) {
+    signed_batch = batch.map((t) => {
+      t.tx.partialSign(sender);
+      return { tx: t.tx, recipient: t.recipient };
+    });
+  }
+  if (isWallet) {
+    for (const item of batch) {
+      await sender.signTransaction(item.tx);
```

Comment:
wallet.signTransaction gives control to wallet and awaits approval, that means, there will be N signatures if im not mistaken, have you tested it with wallet? we should use signAllTransactions and make sure there's only one wallet signature approval for the user